bpf/include/bpf/builtins.h

```diff
@@ -6,17 +6,240 @@
 
 #include "compiler.h"
 
+#ifndef __non_bpf_context
+
 #ifndef lock_xadd
 # define lock_xadd(P, V)	((void) __sync_fetch_and_add((P), (V)))
 #endif
 
-#ifndef memset
-# define memset(S, C, N)	__builtin_memset((S), (C), (N))
+#define __it(x, op) (x -= sizeof(__u##op))
+
+static __always_inline __maybe_unused void
+__bpf_memset_builtin(void *d, __u8 c, __u64 len)
+{
+	/* Everything non-zero or non-const (currently unsupported) as c
+	 * gets handled here.
+	 */
+	__builtin_memset(d, c, len);
+}
+
+static __always_inline void __bpf_memzero(void *d, __u64 len)
+{
+#if __clang_major__ >= 10
+	const __u64 s = 0;
```

Comment:
Couldn't we get rid of `s` since it's always zero and simplify the macro?

Reply:
Ah yeah, code generation wise, it shouldn't make a difference though.